mycore-pi/src/main/java/org/mycore/pi/MCRPICronJob.java

```diff
@@ -0,0 +1,98 @@
+package org.mycore.pi;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import javax.servlet.ServletContext;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mycore.common.MCRException;
+import org.mycore.common.MCRSystemUserInformation;
+import org.mycore.common.events.MCRStartupHandler;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.pi.backend.MCRPI;
+import org.mycore.pi.exceptions.MCRIdentifierUnresolvableException;
+import org.mycore.pi.urn.MCRDNBURN;
+import org.mycore.pi.urn.MCRDNBURNParser;
+import org.mycore.pi.urn.MCRURNUtils;
+import org.mycore.util.concurrent.MCRFixedUserCallable;
+
+public class MCRPICronJob extends TimerTask implements MCRStartupHandler.AutoExecutable {
+
+    private static final int CHECK_URN_THREAD_COUNT = 5;
```

Comment:
magic number: 5

Reply:
Its not a magic number, the name clear stating for what this number is.

Reply:
It says what is it for, but not why it is "5"

Reply:
What about deriving the value of `Runtime.getRuntime().availableProcessors()`:
```
(int) Math.max(1, Runtime.getRuntime().availableProcessors() / 4);
```

Reply:
availableProcessors doesnt matter because the Bottleneck is the request to the dnb. The value 5 works good with one processor and with 16, because the most time the thread waits for the connection.